src/healthRecord/healthRecord.controller.ts

```diff
@@ -24,27 +24,76 @@ export class HealthRecordController {
-  @Get('/:elderlyId/:healthRecordName')
+  @ApiParam({ name: 'elderlyId', type: String, description: 'id of elderly interested' })
+  @Get('/table/:elderlyId/:healthRecordName')
   async getHealthDataByCaretaker(
-    @Param('elderlyId') elderlyId,
+    @Param('elderlyId') elderlyId: string,
     @Param('healthRecordName') healthRecordName: string,
     @Request() req
   ): Promise<HealthRecordTableDto> {
     const caretakerId = req.user.uid;
-    await this.userService.checkRelationship(elderlyId, caretakerId);
-    return await this.healthRecordService.getHealthData(elderlyId, healthRecordName);
+    await this.userService.checkRelationship(Number(elderlyId), Number(caretakerId));
+    await this.healthRecordService.checkHealthRecordExist(Number(elderlyId), healthRecordName);
+    return await this.healthRecordService.getHealthTable(Number(elderlyId), healthRecordName);
+  }
+
+  @UseGuards(ElderlyGuard)
+  @ApiOkResponse({ type: HealthRecordAnalyticsDto })
+  @ApiNotFoundResponse({ description: 'Health Column Name Not Found' })
+  @ApiNotFoundResponse({ description: 'Health Record Name Not Found' })
+  @ApiParam({ name: 'healthRecordName', type: String, description: 'health record name user wants to find' })
+  @ApiParam({ name: 'columnName', description: 'column name in health record user wants to query' })
+  @ApiParam({ name: 'timespan', type: String, enum: Timespan, description: 'timespan of data user wants to query' })
+  @Get('analytics/:healthRecordName/:columnName/:timespan')
+  async getHealthAnalyticsByElderly(
+    @Param('healthRecordName') healthRecordName: string,
+    @Param('columnName') columnName: string,
+    @Param('timespan') timespan: Timespan = Timespan.Year,
+    @Request() req
+  ): Promise<HealthRecordAnalyticsDto> {
+    const elderlyId = req.user.uid;
+    await this.healthRecordService.checkHealthRecordExist(Number(elderlyId), healthRecordName);
+    await this.healthRecordService.checkHealthColumnExist(Number(elderlyId), healthRecordName, columnName);
+    return await this.healthRecordService.getHealthAnalytics(Number(elderlyId), healthRecordName, columnName, timespan);
+  }
+
+  @UseGuards(CaretakerGuard)
+  @ApiOkResponse({ type: HealthRecordAnalyticsDto })
+  @ApiNotFoundResponse({ description: 'Elderly Does Not Exist' })
+  @ApiNotFoundResponse({ description: 'Health Column Name Not Found' })
+  @ApiNotFoundResponse({ description: 'Health Record Name Not Found' })
+  @ApiParam({ name: 'elderlyId', type: String, description: 'id of elderly user wants to query' })
+  @ApiParam({ name: 'healthRecordName', type: String, description: 'health record name user wants to find' })
+  @ApiParam({ name: 'columnName', type: String, description: 'column name in health record user wants to query' })
+  @ApiParam({ name: 'timespan', type: String, enum: Timespan, description: 'timespan of data user wants to query' })
+  @Get('analytics/:elderlyId/:healthRecordName/:columnName/:timespan')
```

Comment:
I think every controller need this `@UsePipes(new ValidationPipe({ whitelist: true }))` to apply class validator

Reply:
For **Get** method , which has no body to pass, I don't find any purpose of using `@UsePipe`. But feel free to correct me of I am wrong.